lib/PluginFindRecord/PluginFindRecordModal.js

```diff
@@ -190,13 +196,21 @@ class PluginFindRecordModal extends React.Component {
     }
 
     const mixedResultsFormatter = {
-      isChecked: record => (
-        <Checkbox
-          type="checkbox"
-          checked={Boolean(checkedMap[record.id])}
-          onChange={() => this.toggleRecord(record)}
-        />
-      ),
+      isChecked: record => {
+        const value = getRecordLabel?.(record) ?? findIndex(records, ['id', record.id]);
+
+        return (
+          <Checkbox
+            type="checkbox"
+            checked={Boolean(checkedMap[record.id])}
+            onChange={() => this.toggleRecord(record)}
+            aria-label={intl.formatMessage(
```

Comment:
it's unreadable, please extract monster lines to separate vars always for more code readability